Bio/PDB/MMCIFParser.py

```diff
@@ -67,6 +67,49 @@ def get_structure(self, structure_id, filename):
 
     # Private methods
 
+    def _mmcif_get(self, key, dict, deflt):
```

Comment:
I'd refactor the function to use a default parameter for default, and avoid using `dict` as a variable name and accessing the key twice:

```
def __mmcif_get(self, key, mmcif_dict, default=None):
    val = mmcif_dict.get(key, default)
    try:
        item = val[0]
    except (TypeError, IndexError):
        pass
    else:
        if item != '?':
            return item
    return val
```

Reply:
done, slightly different approach but my previous effort was to match dictionary .get() syntax.